bitwarden_license/src/Commercial.Core/SecretsManager/AuthorizationHandlers/ServiceAccounts/ServiceAccountAuthorizationHandler.cs

```diff
@@ -169,4 +172,19 @@ public class
             context.Succeed(requirement);
         }
     }
+
+    private async Task CanReadEventsAsync(AuthorizationHandlerContext context,
+        ServiceAccountOperationRequirement requirement, ServiceAccount resource)
+    {
+        var (accessClient, userId) =
+            await _accessClientQuery.GetAccessClientAsync(context.User, resource.OrganizationId);
+        var access =
+            await _serviceAccountRepository.AccessToServiceAccountAsync(resource.Id, userId,
+                accessClient);
+
+        if (access.Write)
```

Comment:
Excuse my (potential, most likely) ignorance: while having limited exposure to the SM permission authorization structure, it feels slightly odd (confusing?) to have a method called `CanReadEventsAsync` but to succeed the check you validate the `Write` property. Just wanted to double check this is the expected outcome?

Reply:
Thanks for the explanation!